src/include/ipebitmap.h

```diff
@@ -77,16 +80,20 @@ namespace ipe {
     inline int objNum() const;
     inline void setObjNum(int objNum) const;
 
-    std::pair<Buffer, Buffer> embed() const;
+    std::pair<Buffer, Buffer> getEmbedData() const;
 
     inline bool operator==(const Bitmap &rhs) const;
     inline bool operator!=(const Bitmap &rhs) const;
     inline bool operator<(const Bitmap &rhs) const;
 
     static const char *readJpegInfo(FILE *file, int &width, int &height,
 				    Vector &dotsPerInch, uint32_t &flags);
-    static Bitmap readJpeg(const char *fname, Vector &dotsPerInch, const char * &errmsg);
-    static Bitmap readPNG(const char *fname, Vector &dotsPerInch, const char * &errmsg);
+    static Bitmap readJpeg(const char *fname, Vector &dotsPerInch,
```

Comment:
You would make my life easier reading your MR if you don't let your editor make formatting changes.

Reply:
Yes, GH is unfortunately pretty bad at ignoring whitespace in changes, there is a switch which makes [things slightly better](https://github.blog/news-insights/product-news/ignore-white-space-in-code-review/). My main issue was that some lines have mixed tabs and spaces like [line 87 right above](https://github.com/otfried/ipe/pull/548/files/77e662b3a8a91d0c5e7eabade7efce481ad71fda#diff-29a99e6063e1f7afbfcf174c78ef7dd0ce535d54fac105cfb35d77996ce1aa93L87) or in the screenshot below, I tried to fix these issues only locally when I touched some function:
![image](https://github.com/user-attachments/assets/0efc0cca-4908-49c6-b5f6-a8b99ef06b46)
Additionally, I was unsure when/how to wrap long lines and whether to wrap opening braces or not.

Have you thought about setting up a [clang-format config file](https://releases.llvm.org/18.1.8/tools/clang/docs/ClangFormatStyleOptions.html) and having the repo autoformatted once (and this maybe even enforced by CI)? This would make the style consistent everywhere and would also help new contributors which don't know which style you like or that have overeager IDEs. There's nice [online](https://clang-format-configurator.site/) [tools](https://zed0.co.uk/clang-format-configurator/) that help you configure the right style

Reply:
In C++, mixing tabs and spaces it not considered harmful :grimacing:  In fact this is how Emacs' C++ mode formats the code.

I had been planning for a long time to start using `clang-format`, because the Ipe coding style is so different from what I use at work.  I've now done it :grin:   Thanks for prodding me to finally take action!